version: '2.1'

volumes:
   db-backups:
   geoserver-data:
   geogig-db-data:
   geo-db-data:

services:
   db:
      image: kartoza/postgis:9.6-2.4
      volumes:
         - geo-db-data:/var/lib/postgresql
      ports:
        - "25432:5432"
      env_file:
      - docker-env/geogig-db.env
      restart: on-failure
      healthcheck:
        test: "exit 0"

   geogig-db:
      image: kartoza/postgis:9.6-2.4
      volumes:
         - geogig-db-data:/var/lib/postgresql
         - db-backups:/backups
      env_file:
         - docker-env/geogig-db.env
      restart: on-failure
      healthcheck:
        test: "exit 0"

   geoserver:
      image: kartoza/geoserver:2.13.0
      volumes:
        - geoserver-data:/opt/geoserver/data_dir
      ports:
        - "8600:8080"
      restart: on-failure
      env_file:
        - docker-env/geoserver.env
      depends_on:
        db:
          condition: service_healthy
        geogig-db:
          condition: service_healthy
      healthcheck:
        test: curl --fail -s http://localhost:8080/ || exit 1
        interval: 1m30s
        timeout: 10s
        retries: 3

   geogig:
      image: kartoza/geogig:latest
      ports:
        - "8182:8080"
      restart: on-failure
      depends_on:
        db:
          condition: service_healthy
        geogig-db:
          condition: service_healthy
      env_file:
        - docker-env/geogig.env
      healthcheck:
        test: "exit 0"

   geogig-db-dbbackups:
      image: kartoza/pg-backup:9.6
      volumes:
        - db-backups:/backups
      env_file:
      - docker-env/geogig-db-backups.env
      restart: on-failure
      depends_on:
        geogig-db:
          condition: service_healthy

   db-dbbackups:
     image: kartoza/pg-backup:9.6
     volumes:
     - db-backups:/backups
     env_file:
     - docker-env/db-backups.env
     restart: on-failure
     depends_on:
       db:
         condition: service_healthy